src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -295,6 +295,18 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
                 var attributeTypesBuilder = new NamedTypeSymbol[totalAttributesCount];
 
                 Binder.BindAttributeTypes(binders, attributesToBind, this, attributeTypesBuilder, diagnostics);
+                if (MessageID.IDS_FeatureGenericAttributes.GetFeatureAvailabilityDiagnosticInfo(compilation) is { } diagnosticInfo)
```

Comment:
>MessageID.IDS_FeatureGenericAttributes.GetFeatureAvailabilityDiagnosticInfo(compilation) is { } diagnosticInfo [](http://example.com/codeflow?start=20&length=110)

I think I'd prefer we only calculate the diagnostic when needed. Perhaps check `compilation.IsFeatureEnabled(MessageID.IDS_FeatureGenericAttributes)` here instead, and calculate the diagnostic in the `if (IsGenericType)` below.

Reply:
Or perhaps just replace this check with an unconditional call to `MessageID.IDS_FeatureGenericAttributes.CheckFeatureAvailability(...);` inside the loop when we've found a generic attribute.

Reply:
Going with the latter. Thanks.